src/main/java/core/basesyntax/strategy/PurchaseOperation.java

```diff
@@ -0,0 +1,16 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.Map;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void handle(FruitTransaction transaction, Map<String, Integer> inventory) {
+        int currentQuantity = inventory.getOrDefault(transaction.getFruit(), 0);
+        if (currentQuantity < transaction.getQuantity()) {
+            throw new RuntimeException(
+                    "Insufficient quantity for fruit: " + transaction.getFruit());
+        }
+        inventory.put(transaction.getFruit(), currentQuantity - transaction.getQuantity());
+    }
```

Comment:
The `PurchaseOperation` class should be placed in its own package to adhere to the project structure requirements. Consider placing it in the `strategy` package as indicated by the checklist.